pyanaconda/modules/timezone/timezone.py

```diff
@@ -0,0 +1,143 @@
+        self._ntp_enabled = True
+        self._ntp_servers = []
+
+    def publish(self):
+        """Publish the module."""
+        DBus.publish_object(TimezoneInterface(self), MODULE_TIMEZONE_PATH)
+        DBus.register_service(MODULE_TIMEZONE_NAME)
+
+    def ping(self, s):
+        """Ping the module."""
+        log.debug(s)
+        return "TimeModule says hi!"
+
+    @property
+    def kickstart_specification(self):
+        """Return the kickstart specification."""
+        return TimezoneKickstartSpecification
+
+    def process_kickstart(self, data):
+        """Process the kickstart data."""
+        self.set_timezone(data.timezone.timezone)
+        self.set_utc(data.timezone.isUtc)
+        self.set_ntp(not data.timezone.nontp)
+        self.set_ntp_servers(data.timezone.ntpservers)
+
+    def generate_kickstart(self):
+        """Return the kickstart string."""
+        data = self.get_kickstart_data()
+        data.timezone.timezone = self.timezone
+        data.timezone.isUtc = self.is_utc
+        data.timezone.nontp = not self.ntp_enabled
+
+        if self.ntp_enabled:
+            data.timezone.ntpservers = list(self.ntp_servers)
+
+        return str(data)
+
+    @signal
+    def module_changed(self):
+        """Signal that module might have changed.
+
+        If someone changes properties of this module
+        on the python level (and not from DBus), he
+        should emit this signal once he is done.
+        """
+        pass
+
+    @property
+    def timezone(self):
+        """Return the timezone."""
+        return self._timezone
+
+    @signal
+    def timezone_changed(self):
+        """Signal for timezone."""
+        pass
+
+    @timezone_changed.trigger
+    def set_timezone(self, timezone):
+        """Set the timezone."""
+        self._timezone = timezone
+
+    @property
+    def is_utc(self):
+        """Is the hardware clock set to UTC?"""
+        return self._is_utc
+
+    @signal
+    def is_utc_changed(self):
+        """Signal for is utc."""
+        pass
+
+    @is_utc_changed.trigger
+    def set_utc(self, is_utc):
+        """Set if the hardware clock is set to UTC."""
+        self._is_utc = is_utc
+
+    @property
+    def ntp_enabled(self):
+        """Enable automatic starting of NTP service."""
+        return self._ntp_enabled
+
+    @signal
+    def ntp_enabled_changed(self):
+        """Signal for ntp enabled."""
+        pass
+
+    @ntp_enabled_changed.trigger
+    def set_ntp(self, ntp_enabled):
+        """Enable or disable automatic starting of NTP service."""
+        self._ntp_enabled = ntp_enabled
+
+    @property
+    def ntp_servers(self):
+        """Return a list of NTP servers."""
+        return self._ntp_servers
+
+    @signal
+    def ntp_servers_changed(self):
+        """Signal for ntp servers."""
+        pass
+
+    @ntp_servers_changed.trigger
+    def set_ntp_servers(self, servers):
+        """Set NTP servers."""
+        self._ntp_servers = list(servers)
```

Comment:
Is there a way how to call `set_ntp_servers` or other trigger method without pulling a trigger? You know it could be that we don't want to announce new value. Maybe it doesn't changed at all.

Reply:
No, but you can always remove the decorator and emit the signal manually from the method, or you can set self._ntp_servers directly if you are in the module, or you can provide a method that sets self._ntp_servers directly without pulling a trigger. I think that there are a lot of ways how to deal with the special cases.

Reply:
OK, I agree there are ways how to avoid triggering so it could be this solution. However we still need to decide on `signal` vs `Signal`.